src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+    {
+        private const int TimeoutWaiting = 10_000;
+
+        private readonly SpiDevice _spiDevice;
+        private readonly GpioController _gpioController;
+        private bool _shouldDispose;
+        private int _pinBusy;
+        private int _pinNss;
+
+        private static List<SelectedPiccInformation> _activeSelected = new List<SelectedPiccInformation>();
+
+        /// <summary>
+        /// A radio Frequence configuration element size is 5 bytes
+        /// Byte 1 = Register Address
+        /// next 4 bytes = data of the register
+        /// </summary>
+        public const int RadioFrequenceConfiguraitonSize = 5;
+
+        /// <summary>
+        /// PN532 SPI Clock Frequency
+        /// </summary>
+        public const int SpiClockFrequency = 7_000_000;
+
+        /// <summary>
+        /// Only SPI Mode supported is Mode0
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+
+        /// <summary>
+        /// The Log level
+        /// </summary>
+        public LogLevel LogLevel
+        {
+            get { return LogInfo.LogLevel; }
+            set { LogInfo.LogLevel = value; }
+        }
+
+        /// <summary>
+        /// The location to log the info
+        /// </summary>
+        public LogTo LogTo
+        {
+            get { return LogInfo.LogTo; }
+            set { LogInfo.LogTo = value; }
+        }
+
+        /// <summary>
+        /// Create a PN5180 RFID/NFC reader
+        /// </summary>
+        /// <param name="spiDevice">The SPI device</param>
+        /// <param name="pinBusy">The pin for the busy line</param>
+        /// <param name="pinNss">The pin for the SPI select line. This has to be handle differently than thru the normal process as PN5180 has a specific way of working</param>
+        /// <param name="gpioController">A GPIO controller, null will use a default one</param>
+        /// <param name="shouldDispose">Dispose the SPI and the GPIO controller at the end if true</param>
+        /// <param name="logLevel">The log level</param>
+        public Pn5180(SpiDevice spiDevice, int pinBusy, int pinNss, GpioController gpioController = null, bool shouldDispose = true, LogLevel logLevel = LogLevel.None)
+        {
+            LogLevel = logLevel;
+
+            LogInfo.Log($"Opening PN5180, pin busy: {pinBusy}, pin NSS: {pinNss}", LogLevel.Debug);
+            _spiDevice = spiDevice;
+            _gpioController = gpioController ?? new GpioController(PinNumberingScheme.Logical);
+            _shouldDispose = shouldDispose;
+            _pinBusy = pinBusy;
+            _pinNss = pinNss;
+            _gpioController.OpenPin(_pinBusy, PinMode.Input);
+            _gpioController.OpenPin(_pinNss, PinMode.Output);
+
+            // Check the version
+            var (product, firmware, eeprom) = GetVersion();
+            if ((product.Major == 0) || (firmware.Major == 0) || (eeprom.Major == 0))
```

Comment:
should this be `&&` or `>= someVersion`? I suspect 1.0.0 should not create an exception, right?

Reply:
it should actually. Versions numbers are not 0. they have lots of variety but not 0. Those are 3 different versions, so if any is 0, then for sure it's not a good version. It's not that easy neither to test something >= for any of them. All up, it's a very basic test just to make sure that there is a basic one.

Reply:
we can fix it in the future if we got it slightly wrong 😄